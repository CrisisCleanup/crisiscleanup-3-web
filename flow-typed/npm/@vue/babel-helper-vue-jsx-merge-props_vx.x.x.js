// flow-typed signature: d7a23c53f4091ef2bf1df8afa5eb43c4
// flow-typed version: <<STUB>>/@vue/babel-helper-vue-jsx-merge-props_v^1.0.0/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-helper-vue-jsx-merge-props'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-helper-vue-jsx-merge-props' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-helper-vue-jsx-merge-props/dist/helper.js' {
  declare module.exports: $Exports<'@vue/babel-helper-vue-jsx-merge-props/dist/helper'>;
}
