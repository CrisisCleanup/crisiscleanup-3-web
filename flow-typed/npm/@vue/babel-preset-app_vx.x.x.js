// flow-typed signature: 6ba91a92845d9a556a5bcdcc4f033fed
// flow-typed version: <<STUB>>/@vue/babel-preset-app_v^4.3.1/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-app'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-app' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-app/polyfillsPlugin' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-app/index' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/index.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app'>;
}
declare module '@vue/babel-preset-app/polyfillsPlugin.js' {
  declare module.exports: $Exports<'@vue/babel-preset-app/polyfillsPlugin'>;
}
