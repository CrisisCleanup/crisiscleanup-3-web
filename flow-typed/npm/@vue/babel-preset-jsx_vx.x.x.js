// flow-typed signature: eeecaeff29e4a3a6c63c4fe7e16a0eb3
// flow-typed version: <<STUB>>/@vue/babel-preset-jsx_v^1.1.2/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/babel-preset-jsx'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/babel-preset-jsx' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/babel-preset-jsx/dist/plugin.cjs' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/babel-preset-jsx/dist/plugin.cjs.js' {
  declare module.exports: $Exports<'@vue/babel-preset-jsx/dist/plugin.cjs'>;
}
