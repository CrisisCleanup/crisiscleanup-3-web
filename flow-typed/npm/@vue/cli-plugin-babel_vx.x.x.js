// flow-typed signature: 4cf1a98e80b05c0a167e013d64bc9645
// flow-typed version: <<STUB>>/@vue/cli-plugin-babel_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-babel'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-babel' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/codemods/usePluginPreset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-babel/preset' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/customConfig.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/default.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/default.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/doubleQuote.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/require.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/require.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.input'>;
}
declare module '@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/__testfixtures__/templateLiteral.output'>;
}
declare module '@vue/cli-plugin-babel/codemods/usePluginPreset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/codemods/usePluginPreset'>;
}
declare module '@vue/cli-plugin-babel/generator.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/generator'>;
}
declare module '@vue/cli-plugin-babel/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel'>;
}
declare module '@vue/cli-plugin-babel/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/migrator'>;
}
declare module '@vue/cli-plugin-babel/preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-babel/preset'>;
}
