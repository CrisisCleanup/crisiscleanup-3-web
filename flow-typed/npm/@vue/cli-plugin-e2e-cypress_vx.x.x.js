// flow-typed signature: 122252e4d027555936ded476aff636b4
// flow-typed version: <<STUB>>/@vue/cli-plugin-e2e-cypress_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-e2e-cypress'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-e2e-cypress' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-e2e-cypress/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-e2e-cypress/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-e2e-cypress/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/_eslintrc'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/plugins'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/specs/test'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/commands'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/generator/template/tests/e2e/support'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress'>;
}
declare module '@vue/cli-plugin-e2e-cypress/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-e2e-cypress/ui'>;
}
