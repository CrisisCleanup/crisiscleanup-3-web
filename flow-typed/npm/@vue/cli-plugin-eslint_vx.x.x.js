// flow-typed signature: 190fd4925231c8aa9cbfe00ac40be4f1
// flow-typed version: <<STUB>>/@vue/cli-plugin-eslint_v^4.4.6/flow_v0.129.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-eslint'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-eslint' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-eslint/eslintDeps' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/eslintOptions' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/lint' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/migrator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/prompts' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/configDescriptor' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-eslint/ui/taskDescriptor' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-eslint/eslintDeps.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintDeps'>;
}
declare module '@vue/cli-plugin-eslint/eslintOptions.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/eslintOptions'>;
}
declare module '@vue/cli-plugin-eslint/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/generator'>;
}
declare module '@vue/cli-plugin-eslint/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint'>;
}
declare module '@vue/cli-plugin-eslint/lint.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/lint'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/migrator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/migrator'>;
}
declare module '@vue/cli-plugin-eslint/prompts.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/prompts'>;
}
declare module '@vue/cli-plugin-eslint/ui/configDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/configDescriptor'>;
}
declare module '@vue/cli-plugin-eslint/ui/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui'>;
}
declare module '@vue/cli-plugin-eslint/ui/taskDescriptor.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-eslint/ui/taskDescriptor'>;
}
