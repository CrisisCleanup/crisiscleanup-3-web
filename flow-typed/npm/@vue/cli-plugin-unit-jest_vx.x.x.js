// flow-typed signature: 345a830bc132c538d84f7a5ce68b6d33
// flow-typed version: <<STUB>>/@vue/cli-plugin-unit-jest_v^4.4.6/flow_v0.131.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@vue/cli-plugin-unit-jest'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@vue/cli-plugin-unit-jest' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@vue/cli-plugin-unit-jest/generator' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/default/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/presets/typescript/jest-preset' {
  declare module.exports: any;
}

declare module '@vue/cli-plugin-unit-jest/ui' {
  declare module.exports: any;
}

// Filename aliases
declare module '@vue/cli-plugin-unit-jest/generator/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator'>;
}
declare module '@vue/cli-plugin-unit-jest/generator/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator'>;
}
declare module '@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/generator/template/tests/unit/example.spec'>;
}
declare module '@vue/cli-plugin-unit-jest/index' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest'>;
}
declare module '@vue/cli-plugin-unit-jest/index.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest'>;
}
declare module '@vue/cli-plugin-unit-jest/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/default/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/default/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/no-babel/esmoduleTransformer'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/no-babel/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/typescript-and-babel/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/presets/typescript/jest-preset.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/presets/typescript/jest-preset'>;
}
declare module '@vue/cli-plugin-unit-jest/ui.js' {
  declare module.exports: $Exports<'@vue/cli-plugin-unit-jest/ui'>;
}
