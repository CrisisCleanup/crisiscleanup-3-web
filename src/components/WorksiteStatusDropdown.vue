<template>
  <v-popover
    popover-class="popover menu"
    placement="bottom-end"
    class="text-xs"
  >
    <div
      class="badge-holder rounded-lg"
      :class="size === 'sm' ? 'px-1' : 'px-2'"
      :style="dropdownStyle"
    >
      <div
        v-if="useIcon"
        class="case-svg-container mr-1 flex"
        ref="svgContainer"
        v-html="workTypeImage"
      ></div>
      <div class="tooltip-target" v-if="!hideName">
        {{ currentWorkType.status | getStatusName }}
      </div>
      <font-awesome-icon
        class="tooltip-target"
        :class="size === 'sm' ? '' : 'mx-1'"
        size="sm"
        icon="chevron-down"
      />
    </div>
    <template #popover>
      <div
        class="
          bg-white
          border
          outline-none
          h-84
          w-56
          overflow-auto
          tooltip-content
        "
        @keyup="nextItem"
      >
        <div
          v-for="status in displayStatuses"
          :key="`${status.id}`"
          :value="status.status"
          class="cursor-pointer py-1 hover:bg-crisiscleanup-light-grey"
          :class="{ selected: currentItem === status.selectionKey }"
        >
          <div
            v-close-popover
            class="badge-holder text-xs"
            @click="
              () => {
                $emit('input', status.status);
              }
            "
          >
            <badge class="mx-1" :color="getColorForStatus(status.status)" />
            <div>{{ status.status_name_t }}</div>
          </div>
        </div>
      </div>
    </template>
  </v-popover>
</template>

<script>
import { getColorForStatus, getWorkTypeImage } from '@/filters';
import { mapState } from 'vuex';
import { WorksitesMixin } from '@/mixins';

export default {
  name: 'WorksiteStatusDropdown',
  mixins: [WorksitesMixin],
  props: {
    currentWorkType: {
      type: Object,
      default: () => {
        return {};
      },
    },
    useIcon: {
      type: Boolean,
      default: false,
    },
    hideName: {
      type: Boolean,
      default: false,
    },
    phase: {
      type: Number,
      default: null,
    },
    iconSize: {
      type: Number,
      default: 0,
    },
    size: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      getColorForStatus,
      getWorkTypeImage,
      currentItem: 1,
    };
  },
  computed: {
    ...mapState('enums', ['statuses']),
    displayStatuses() {
      return this.statuses
        .filter((status) =>
          this.phase ? status.phases.includes(this.phase) : true,
        )
        .map((status, index) => {
          return {
            ...status,
            selectionKey: index + 1,
          };
        });
    },
    workTypeImage() {
      if (this.iconSize) {
        return this.getWorktypeSVG(this.currentWorkType, this.iconSize);
      }
      return this.getWorkTypeImage(this.currentWorkType);
    },
    dropdownStyle() {
      return {
        color: getColorForStatus(
          this.currentWorkType.status,
          Boolean(this.currentWorkType.claimed_by),
        ),
        backgroundColor: `${getColorForStatus(
          this.currentWorkType.status,
          Boolean(this.currentWorkType.claimed_by),
        )}3D`,
      };
    },
  },
  mounted() {
    document.addEventListener('keyup', this.nextItem);
    this.$nextTick(() => this.setSVGStyles());
  },
  methods: {
    nextItem(e) {
      if (e.keyCode === 38 && this.currentItem > 1) {
        this.currentItem -= 1;
      } else if (
        e.keyCode === 40 &&
        this.currentItem < this.displayStatuses.length
      ) {
        this.currentItem += 1;
      }
    },
    setSVGStyles() {
      if (!this.$refs.svgContainer) return;
      if (this.iconSize) {
        this.$refs.svgContainer.style.minHeight = `${this.iconSize}px`;
        this.$refs.svgContainer.style.minWidth = `${this.iconSize}px`;
      }
    },
  },
};
</script>

<style>
.status-dropdown {
}
</style>

<style scoped>
.badge-holder {
  @apply flex items-center cursor-pointer;
}
</style>
