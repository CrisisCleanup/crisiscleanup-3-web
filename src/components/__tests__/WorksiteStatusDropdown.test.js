/**
 *
 * Tests for WorksiteStatusDropdown
 *
 * Components
 */

import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import WorksiteStatusDropdown from '../WorksiteStatusDropdown.vue';
const statuses = [
  {
    id: 1,
    status_name_t: 'status.open_unassigned',
    substatus_name_t: 'substatus.unassigned',
    description_t: 'statusDescription.open_unassigned',
    status: 'open_unassigned',
    primary_state: 'open',
    substatus: 'unassigned',
    claimed_class: 'open-unassigned-claimed',
    unclaimed_class: 'open-unassigned-unclaimed',
    completed_by_anybody: '0.00',
    completed_by_ccu: '0.00',
    completed_db: '0.00',
  },
  {
    id: 2,
    status_name_t: 'status.open_assigned',
    substatus_name_t: 'substatus.assigned',
    description_t: 'statusDescription.open_assigned',
    status: 'open_assigned',
    primary_state: 'open',
    substatus: 'assigned',
    claimed_class: 'open-assigned-claimed',
    unclaimed_class: 'open-assigned-unclaimed',
    completed_by_anybody: '0.25',
    completed_by_ccu: '0.25',
    completed_db: '0.25',
  },
];

const localVue = createLocalVue();

localVue.use(Vuex);
// mock filter
localVue.filter(
  'getStatusName',
  jest.fn(() => 'status.open_unassigned'),
);

const mockWorkType = {
  id: 1,
  name_t: 'workType.ash',
  description_t: 'workTypeDescription.ash',
  key: 'ash',
  file_prefix: 'Ash',
};

const mountWithOptions = (store, props = {}) =>
  shallowMount(WorksiteStatusDropdown, {
    stubs: {
      'font-awesome-icon': true,
      'a-select': true,
      'v-popover': true,
      badge: true,
    },
    propsData: {
      currentWorkType: mockWorkType,
      ...props,
    },
    localVue,
    store,
  });

describe('WorksiteStatusDropdown', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        enums: {
          namespaced: true,
          state: {
            statuses,
          },
        },
      },
    });
  });

  it('should not log any errors', () => {
    const spy = jest.spyOn(global.console, 'error');
    mountWithOptions(store);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should change current item on keypress', () => {
    const wrapper = mountWithOptions(store);
    // mock sanity check
    expect(wrapper.vm.statuses.length).toBe(2);
    expect(wrapper.vm.currentItem).toBe(1);
    wrapper.vm.nextItem({ keyCode: 40 });
    expect(wrapper.vm.currentItem).toBe(2);
    wrapper.vm.nextItem({ keyCode: 40 });
    expect(wrapper.vm.currentItem).toBe(2);
    wrapper.vm.nextItem({ keyCode: 38 });
    expect(wrapper.vm.currentItem).toBe(1);
  });

  it('should render correctly and match snapshot', () => {
    const wrapper = mountWithOptions(store);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly and match snapshot when using icons', () => {
    const wrapper = mountWithOptions(store, { useIcon: true });
    expect(wrapper.element).toMatchSnapshot();
  });
});
