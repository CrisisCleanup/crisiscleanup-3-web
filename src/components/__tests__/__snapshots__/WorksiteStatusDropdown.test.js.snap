// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorksiteStatusDropdown should render correctly and match snapshot 1`] = `
<v-popover-stub
  class="text-xs"
  placement="bottom-end"
  popover-class="popover menu"
>
  <div
    class="badge-holder rounded-lg px-2"
    style="color: rgb(208, 2, 27);"
  >
    <!---->
     
    <div>
      status.open_unassigned
    </div>
     
    <font-awesome-icon-stub
      class="tooltip-target mx-1"
      icon="chevron-down"
      size="sm"
    />
  </div>
</v-popover-stub>
`;

exports[`WorksiteStatusDropdown should render correctly and match snapshot when using icons 1`] = `
<v-popover-stub
  class="text-xs"
  placement="bottom-end"
  popover-class="popover menu"
>
  <div
    class="badge-holder rounded-lg px-2"
    style="color: rgb(208, 2, 27);"
  >
    <div
      class="case-svg-container mr-1 flex"
    />
     
    <div>
      status.open_unassigned
    </div>
     
    <font-awesome-icon-stub
      class="tooltip-target mx-1"
      icon="chevron-down"
      size="sm"
    />
  </div>
</v-popover-stub>
`;
